### ISSUE TITLE
Brainstorm

### ISSUE THREAD

**Issue by evandroeisinger** [evandroeisinger/dungeon-and-algorithms#1]
Do a brainstorm to define what will be the game

**Comment by evandroeisinger** [comment]
Dungeon and Algorithms - Um jogo de terminal onde um mestre cria uma dungeon, está poderá ser desafiada por qualquer algoritmo. O objetivo é simples: chegar até o final com vida. Durante a dungeon os algoritmos irão enfrentar diversos obstáculos e inimigos, cabe a eles decidirem quais serão os seus movimentos. No início de cada dungeon os algoritmos terão que distribuir os seus pontos de habilidade, estes pontos influenciaram na eficácia dos seus movimentos. Toda falha custará pontos de vida e estes pontos não serão recuperados. Aquele que conseguir chegar até o fim da dungeon, e estiver com o mais pontos de vidas, será o ganhador. Apenas o ganhador receberá um bônus de pontos de habilidade, este bônus será acumulativo e poderá ser usado nas próximas dungeons.

- atributos: força, agilidade, inteligência, resistência...
- desafios: abrir portas, decifrar enigmas, pular obstáculos, derrotar inimigos;
- elementos: terra, metal, madeira, ferro, cristal...
- criaturas: cachorro, leão, zumbi, mago...
- bônus-de-jogada: quanto mais rápido (tempo de resp.) o algoritmo resolver a sua jogada, maior será o seu bônus de movimento.
- bônus-de-partida: quanto mais vitórias o algoritmo ter em sua trajetória na dungeon, maior será o seu bônus ao ganhar.

**Comment by marceloboeira** [comment]
Algumas dúvidas, no início de cada partida será tudo zerado, ou o jogador poderá ter um perfil e ir acumulando pontos ao longo do tempo? tipo pra ir melhorando o personagem ...

**Comment by evandroeisinger** [comment]
Exatamente! Acho que não me expressei muito bem quando comentei que: Apenas o ganhador receberá um bônus de pontos de habilidade, este bônus será acumulativo e poderá ser usado nas próximas dungeons.

Obs: O que coloquei ali em cima é apenas um insight que tive, para que eu não me esquecesse :ghost:

**Comment by juanibiapina** [comment]
Achei muito complicado. Queria pensar em algo muito simples que desse pra ser feito no nosso pouquíssimo tempo livre. Vou tentar criar uma lista de requerimentos que eu acho importantes, ai vocês podem ver se concordam e a gente evolui.

1. Simples

    Não deve ter regras complicadas e muitas situações pra calcular. Isso significa dizer que pode ser mais estratégico e menos tático.

2. Todo mundo que quiser pode participar ao mesmo tempo

3. Extensível

    Deve ser fácil de inserir novos desafios, caso nossos algoritmos já estejam muito bons

4. Estimule conversas

    Seria legal se o jogo fizesse a gente conversar uns com os outros sobre o que aconteceu, potencialmente discutindo estratégias.

**Comment by evandroeisinger** [comment]
Estava pensando em algo simples, onde o objetivo é criar um algoritmo que consiga sair de um labirinto o mais rápido possível. O labirinto pode ser gerado manualmente ou até mesmo dinamicamente :+1:
